quinn-proto/src/connection.rs

```diff
@@ -2298,28 +2307,6 @@ impl From<TransportError> for ConnectionError {
     }
 }
 
-impl From<ConnectionError> for io::Error {
-    fn from(x: ConnectionError) -> io::Error {
-        use self::ConnectionError::*;
-        match x {
-            TimedOut => io::Error::new(io::ErrorKind::TimedOut, "timed out"),
-            Reset => io::Error::new(io::ErrorKind::ConnectionReset, "reset by peer"),
-            ApplicationClosed { reason } => io::Error::new(
-                io::ErrorKind::ConnectionAborted,
-                format!("closed by peer application: {}", reason),
-            ),
-            ConnectionClosed { reason } => io::Error::new(
-                io::ErrorKind::ConnectionAborted,
-                format!("peer detected an error: {}", reason),
-            ),
-            TransportError { error_code } => {
-                io::Error::new(io::ErrorKind::Other, format!("{}", error_code))
-            }
-            VersionMismatch => io::Error::new(io::ErrorKind::Other, "version mismatch"),
-        }
-    }
-}
-
```

Comment:
This was included as a convenience when interoperating with legacy code written for OS socket APIs. Are we sure we want to make that more difficult?

Reply:
Kept it and added a comment.